.github/workflows/ci.yml

```diff
@@ -45,6 +45,20 @@ jobs:
           mkdir -p $NXF_SINGULARITY_CACHEDIR
           mkdir -p $NXF_SINGULARITY_LIBRARYDIR
 
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install nf-core
+        run: |
+          pip install nf-core
+
+      - name: Download containers
+        # Forcibly download repo on branch hic_opt and download SINGULARITY containers into the CACHE dir if not found
+        run: |
+          nf-core download sanger-tol/treeval --revision hic_opt --compress none -d --force --outdir sanger-treeval --container-cache-utilisation amend --container-system singularity
```

Comment:
This step pull all 35 sif image in advance before the run and it seems help lots . 

After downloading, it seems we should run the pipeline differently according to the document. I never tried the nf-core download myself.

https://nf-co.re/tools#downloading-pipelines-for-offline-use

`nextflow run /path/to/download/nf-core-rnaseq-dev/workflow/ --input mydata.csv --outdir results  # usual parameters here`

Reply:
I can do that.

I've removed the checkout@v4 and how have it running the nf-core downloaded version. Has meant I've had to include another action to get branch names. See more on #234 .